### DISCUSSION TITLE
stories.json

### DISCUSSION THREAD

**Discussion by JohannesAnd** [storybookjs/storybook#12336] [Help] [answered]
Hello,

I am trying to use storybook composition by hosting the storybooks in google cloud storage. I am however getting an error where it looks for a `stories.json` file which does not get created when building storybook. How would I go about making this work? Do I need to create this `stories.json` file?` I've looked at a lot of issues etc. on this repo, but not found a solution yet.

**Comment by shilman** [comment]
You can generate the `stories.json` using `npx sb extract`. @ndelangen @jonniebigodes this needs documentation

**Reply by ndelangen** [reply]
@jonniebigodes is this something that's on your todo-list?

I'm guessing we should mention it on this page:
https://storybook.js.org/docs/react/workflows/storybook-composition

**Reply by jonniebigodes** [reply]
@shilman and @ndelangen yes, this needs documentation and i have in my todo list. A pull request will be up shortly addressing this.

**Reply by JohannesAnd** [reply]
Thanks for the help! I do however have a problem where `npx sb extract` times out every time due to having to install puppeteer.

**Reply by aharris** [reply]
I am also getting the timeout:

```
$ npx sb extract
info connecting to: http://localhost:58292/iframe.html
info installing puppeteer...
TimeoutError: waiting for function failed: timeout 30000ms exceeded
    at new WaitTask (/Users/aharris/.npm/_npx/63375/lib/node_modules/sb/node_modules/puppeteer-core/lib/DOMWorld.js:549:28)
    at DOMWorld.waitForFunction (/Users/aharris/.npm/_npx/63375/lib/node_modules/sb/node_modules/puppeteer-core/lib/DOMWorld.js:454:12)
    at Frame.waitForFunction (/Users/aharris/.npm/_npx/63375/lib/node_modules/sb/node_modules/puppeteer-core/lib/FrameManager.js:657:28)
    at Page.waitForFunction (/Users/aharris/.npm/_npx/63375/lib/node_modules/sb/node_modules/puppeteer-core/lib/Page.js:1141:29)
    at read (/Users/aharris/.npm/_npx/63375/lib/node_modules/sb/node_modules/@storybook/cli/dist/extract.js:30:14)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async extract (/Users/aharris/.npm/_npx/63375/lib/node_modules/sb/node_modules/@storybook/cli/dist/extract.js:87:18) {
  name: 'TimeoutError'
}
```

**Reply by aharris** [reply]
Looks like you need to build prior to running this: 

```
yarn build-storybook && npx sb extract
```

**Reply by JohannesAnd** [reply]
I found the issue. I am building storybook at a publicPath `subStorybooks/xyz` which is configured in the webpack config. As far as I can tell, there are no config for extracting from here. It will serve it at root, and the linked files will not be loaded correctly.

When I set the publicPath to `/` it works fine

**Reply by JohannesAnd** [reply]
Changing the line in the `useLocation` function in `extract.js`
```javascript
const result = `http://localhost:${port}/iframe.html`;
```
to
```javascript
const result = `http://localhost:${port}/subStorybooks/xyz/iframe.html`;
```

seems to solve the issue. Maybe you can expose an option for defining the publicPath for extract?

**Reply by jonniebigodes** [reply]
@JohannesAnd i had a chat during this week with @ndelangen and he told me that you can use `sb extract` to a given folder. I've covered it in the pull request. You can use `sb extract storybook-build-directory directory-for-stories-json/` with that you probably can use it for a nested one. Based on what he told me you'll need to add both arguments to make it work. 

See if it helps

Stay safe

**Reply by JohannesAnd** [reply]
That would have worked if the requests from iframe.html were relative. If I serve the code from the subdirectory, the html still tries to read it from the whole path from root. Setting the paths makes the iframe.html (located at root) try to get files from '/subStorybooks/xyz/somefile.something'.

**Comment by lainafoodx** [comment]
metadata.json also fails to be created.  npx sb extract unfortunately does not resolve the issue.

**Reply by shilman** [reply]
I don't think metadata.json is required?

**Reply by ndelangen** [reply]
metadata.json is optional indeed

**Comment by aharris** [comment]
`npx sb extract` resolved my this issue for me.

I will note though that this builds to `./storybook-static` so you need to include that in your `start-storybook` command in order for it to work.

`start-storybook -p 6006 -s ./storybook-static`

**Reply by aharris** [reply]
Is there anyway to get this to run from the dev server rather than needed to extract with every file change?

**Reply by ndelangen** [reply]
This should be pretty easy to implement, I'd be happy to assist a bit in a zoom call if you're interested in contributing this feature:

https://calendly.com/ndelangen/storybook

**Comment by jonniebigodes** [comment]
Just created a pull request to address this discussion in [here](https://github.com/storybookjs/storybook/pull/12510). If all parties involved in this discussion are ok i would love to get some feedback in the pull request to see if you're all ok with it, so that we can merge it, or continue to work on it.

Stay safe all of you

**Comment by Meemaw** [comment]
Is it correct that `npx sb extract` is required to make Storybook composition work? The documentation is a bit unclear on that. It mentions it is used as an optimisation, but it seems composition does not work without it.

Anyway, trying to do `npx sb extract` in CI (Github Actions) before uploading assets consistently throws an error:

```js
info connecting to: http://localhost:43277/iframe.html
info installing puppeteer...
TimeoutError: waiting for function failed: timeout 30000ms exceeded
    at new WaitTask (/home/runner/.npm/_npx/4034/lib/node_modules/sb/node_modules/puppeteer-core/lib/DOMWorld.js:549:28)
    at DOMWorld.waitForFunction (/home/runner/.npm/_npx/4034/lib/node_modules/sb/node_modules/puppeteer-core/lib/DOMWorld.js:454:12)
    at Frame.waitForFunction (/home/runner/.npm/_npx/4034/lib/node_modules/sb/node_modules/puppeteer-core/lib/FrameManager.js:657:28)
    at Page.waitForFunction (/home/runner/.npm/_npx/4034/lib/node_modules/sb/node_modules/puppeteer-core/lib/Page.js:1141:29)
    at read (/home/runner/.npm/_npx/4034/lib/node_modules/sb/node_modules/@storybook/cli/dist/extract.js:30:14)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async extract (/home/runner/.npm/_npx/4034/lib/node_modules/sb/node_modules/@storybook/cli/dist/extract.js:87:18)
```

Any ideas @shilman | @ndelangen  ?

**Reply by shilman** [reply]
I believe it's an optimization and that it should still work without, but that might have changed over the course of development. @ndelangen would know best!

**Reply by Meemaw** [reply]
Hmm, I have that should work in that case here: https://storybook.rebrowse.dev/index.html

You can see in network tab that `stories.json` requests fails and nothing appears cc @ndelangen

**Reply by ndelangen** [reply]
your stories don't appear because there's a runtime error:
<img width="514" alt="Screenshot 2020-11-27 at 19 06 38" src="https://user-images.githubusercontent.com/3070389/100475589-cb36ae80-30e3-11eb-9b09-3d555970ba58.png">

**Reply by ndelangen** [reply]
This happens somewhere in you component code :(

**Reply by ndelangen** [reply]
Composition doesn't require **stories.json** if the request fails, that's handled gracefully.

**Reply by Blizard** [reply]
I have the same problem as @Meemaw. There is error during sb extract on CI:

```
Error: Failed to launch the browser process! spawn /data/jenkins/workspace/i-new_feature_storybook-redesign/node_modules/puppeteer-core/.local-chromium/linux-722234/chrome-linux/chrome ENOENT

TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/master/docs/troubleshooting.md

    at onClose (/data/jenkins/workspace/i-new_feature_storybook-redesign/node_modules/puppeteer-core/lib/Launcher.js:750:14)
    at ChildProcess.helper.addEventListener.error (/data/jenkins/workspace/i-new_feature_storybook-redesign/node_modules/puppeteer-core/lib/Launcher.js:741:65)
    at ChildProcess.emit (events.js:198:13)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:246:12)
    at onErrorNT (internal/child_process.js:415:16)
    at process._tickCallback (internal/process/next_tick.js:63:19)
error Command failed with exit code 1.
```

i also need stories.json for storybook composition. It does not work without it.

**Reply by remysabliet** [reply]
Same issue,  this `npx sb extract` command is finishing with timeout error because of **puppeteer**

**Reply by DimmDell** [reply]
Still having the same with storybook 6.1.14, `npx sb extract` finishes with `TimeoutError: waiting for function failed: timeout 30000ms exceeded`. I checked that there are no runtime errors in my SB, so i dunno why that happens.

**Reply by ndelangen** [reply]
@DimmDell Do you have a reproduction you can share so I can debug this?

**Reply by TomWebbio** [reply]
@ndelangen

I literally cannot get any situation where this does not happen, so it should be easy to reproduce. Even generating a new Storybook project it will give a timeout on the Puppeteer install.

To be clear, just generate a new CRA project, run SB init and start it up to see that it's working, then run sb extract and watch it fail.

**Reply by tobi1220** [reply]
if you're struggling with the `waiting for function failed: timeout 30000ms exceeded` error from puppeteer, try having a static version of your storybook build with the script `build-storybook` instead of running `start-storybook` and then try to generate the `stories.json`
e.g.
```
npm run build-storybook
npx sb extract
```

also make sure you're passing the location your static storybook is residing in as an argument if it's not the default path, like referenced [here](https://storybook.js.org/docs/react/workflows/storybook-composition#improve-your-storybook-composition):

```
npx sb extract my-built-storybook-directory my-other-directory/stories.json
```

**Comment by avegancafe** [comment]
Has anyone gotten `npx sb extract` to work locally? I'm trying to point to a local version of storybook via linking to something like `localhost:3001` and it requests for the `stories.json` file and then is permanently stuck in the loading animation for that sub-storybook instance

**Reply by shilman** [reply]
There is an experimental built-in `stories.json` generation. Here's how to use it:
- [ ] Upgrade 6.4-alpha (`npx sb upgrade --prerelease`)
- [ ] Only works on CSF & MDX stories (no `storiesOf`)
- [ ] Set the following experimental flag in your `.storybook/main.js` config:

```js
module.exports = {
  features: {
    buildStoriesJson: true,
  }
};
```

Then when you `yarn build-storybook` it should also generate `stories.json` in the output folder. This stories.json is not exactly the same format as what comes out of `sb extract`, but it seems to work pretty well already even though it's experimental. It will become the default in 6.4.

**Reply by avegancafe** [reply]
Gotya, yeah I just don't want to have to `build` every time I want to use this feature, since it's for pointing at a local version, not a built version. Good to know the requirement of the stories.json file will be only until hopefully the 6.4 release and it can be generated on request though

**Reply by shilman** [reply]
The new version also generates `stories.json` on `yarn storybook`. However, it's currently only on startup and doesn't change as you add/remove stories. The final version in 6.4 should HMR properly.

**Reply by innoveltec** [reply]
I was using storybook v6.3.0 within the nx workspace. Changed the storybook dependencies to v6.4.0-beta3, added the  buildStoriesJson: true property, after that i build my storybook project of my angular library. The build went ok but when running storybook I also have the message 

Oh no! Your Storybook is empty. Possible reasons why:
The glob specified in main.js isn't correct.
No stories are defined in your story files.

When i look inside the dist folder and the stories.json file i see => {"v":3,"stories":{}}

**Reply by shilman** [reply]
Try upgrading to beta.4. I just released an update related to the built in stories.json that should fix some problems. And if it doesn't, any chance you can share a reproduction? I'm actively working on this

**Reply by Tim-arts** [reply]
Well, all I know is that it doesn't fix the:

```
Oh no! Your Storybook is empty. Possible reasons why:
The glob specified in main.js isn't correct.
No stories are defined in your story files.
```
message. Going back to // buildStoriesJson: true,

@innoveltec could you provide a reproduction of the issue? My repo is tied to private packages and some additional  complexity
@shilman iirc you should have the issue for a fresh Angular project

**Reply by innoveltec** [reply]
@Tim-arts Unfortunately my repo is also private company property. I can try to make a reproduction by generating a new nx workspace using the storybook v6.4-beta.4 version, adding an angular lib with dummy component and generate some stories. Let's see if the issue is still there.

**Reply by Tim-arts** [reply]
Just tried with a fresh Angular install, the issue is present right there (with these additional parameters):
```
features: {
    previewCsfV3: true,
    buildStoriesJson: true,
  },
```

![image](https://user-images.githubusercontent.com/10064887/135921035-4e7d7cde-ba40-47a2-a7c8-f372f919e8d7.png)

**Reply by innoveltec** [reply]
@shilman I have created a repro using a nx workspace with an angular library and a button component with some stories. => [https://github.com/innoveltec/my-story](https://github.com/innoveltec/my-story)

Using v6.4.0-beta.4 when not adding the buildStoriesJson property my storybook is running and showing the stories. I am getting a console error because it is trying to fetch the stories.json file which is not there but still it shows my stories.
If i add the property i am getting the storybook is empty message and no stories. 

In my private company repo which i think is having the same setup not putting the buildStoriesJson property is giving me an error in the console and the storybook page keeps loading because it is trying to fetch the stories.json file which is not there i think. So different behaviour then in my reproduction repo which loads the storybook and the stories  regardless of the stories.json file not being there.

**Reply by shilman** [reply]
@innoveltec I am unable to get your reproduction working. it looks like there are no stories in the `.storybook/main.js` file. can you add instructions for how to run it?

@Tim-arts I just created a vanilla angular12 install with SB 6.4-beta.4 and had no issues, with or without the `buildStoriesJson` flag.

**Reply by Tim-arts** [reply]
It might be related to a library you're using in SB that does not do the job correctly on a Windows environment then (I assume you're not on Windows)

**Reply by shilman** [reply]
Is everybody who is experiencing this issue using Windows?

**Reply by innoveltec** [reply]
@shilman my stories are located in the libs/my-ui. There you have the .storybook folder with the main for the my-ui library. Running the storybook for my-ui library can be done with the command ‘nx run my-ui:storybook’ building can be done with ‘nx run my-ui:build-storybook’

**Reply by innoveltec** [reply]
I am on windows.

**Reply by shilman** [reply]
Thanks for providing the repro, and it's working fine on my machine (mac).


I've opened this issue for windows compatibility and will try to track it down this ASAP: https://github.com/storybookjs/storybook/issues/16237

**Reply by istvandesign** [reply]
> @shilman my stories are located in the libs/my-ui. There you have the .storybook folder with the main for the my-ui library. Running the storybook for my-ui library can be done with the command ‘nx run my-ui:storybook’ building can be done with ‘nx run my-ui:build-storybook’

Hi shilman, did you find a solution to generate the stories.json with nx ?

**Reply by shilman** [reply]
You'll need to ask the nx team about that cc @mandarini @juristr

**Comment by Tim-arts** [comment]
Setting up `buildStoriesJson: true` with the latest version (6.4.0 beta) of Storybook makes the interface flickers between the project stories architecture and this message:

```
Oh no! Your Storybook is empty. Possible reasons why:
The glob specified in main.js isn't correct.
No stories are defined in your story files.
```

When you reload the page if you're lucky enough you can see your stories, but most of the time you'll see this message

**Reply by MickL** [reply]
Im at 6.4.0 beta and I dont understand why there is no stories.json and why I should create it manually.

**Reply by alb3rtuk** [reply]
Same. How do we disable this? It seems to work fine without the stories.json being there, I just want get rid of the error.

**Reply by shilman** [reply]
Did you try upgrading to the latest beta?

```
npx sb@next upgrade --prerelease
```

We've fixed a ton of issues in the past couple weeks that were caused by this [massive rewrite of Storybook's core](https://storybook.js.org/blog/storybook-on-demand-architecture/).

If you're still seeing the problem after upgrading and can share a reproduction, I'd be happy to take a look. Thanks! See [how to create a repro](https://storybook.js.org/docs/react/contribute/how-to-reproduce) for more info.

**Reply by Tim-arts** [reply]
Upgrading to beta.11 fixed all issues on my side (Angular project) (enabling buildStoriesJson flag doesn't make the interface flicker anymore and no error in console so far).

**Comment by taismassaro** [comment]
Trying my luck here as I spent the whole afternoon trying to find a good explanation on how to setup Storybook Composition without success.

We're splitting our internal UI library into a monorepo and would like to use composition to put all the separate package stories into one unified Storybook. Basically we have the following structure:
```
- docs/storybook
- packages/framework
- package/components
```
Where we'd like `docs/storybook` to compose the stories from the `packages`.

I'm on the latest version (6.5.9) and still struggling to understand what the steps actually are.

I've added the `refs` object to the `main.js` config file in `docs/storybook` and I'm running `yarn storybook` for all projects. I've tried both pointing the ref url to localhost while running the package Storybook locally and to a deployed version, none of them worked. I always get infinite loading on the UI and a 404 response for the `stories.json` file. 

Could anyone help me understand what I should be doing instead? This is really blocking us at work right now.

**Reply by shilman** [reply]
Please add the following to your `.storybook/main.js`

```js
module.exports = {
  features: {
    buildStoriesJson: true
  }
}
```

If you're using 6.4 or higher, this should build the `stories.json` file. However, AFAIK in 6.x `stories.json` is an optimization, so I'd expect composition to work without it if all your storybooks are running and the `refs` are set up.

**Reply by taismassaro** [reply]
@shilman thanks for the quick response, I added the config as you mentioned, but I still get
```
Access to fetch at 'http://localhost:6007/stories.json' from origin 'http://localhost:6005' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'.
```
although now I get a 200 response for the stories file.

in `docs/storybook` this is the content of `main.js`
```
module.exports = {
  features: {
    buildStoriesJson: true
  },
  stories: [
    '../stories/**/*.stories.mdx',
    '../stories/**/*.stories.@(js|jsx|ts|tsx)'
  ],
  addons: [
    '@storybook/addon-links',
    '@storybook/addon-essentials',
    '@storybook/addon-interactions'
  ],
  framework: '@storybook/react',
  refs: {
    framework: {
      title: 'framework',
      url: 'http://localhost:6007'
    }
  }
}
```

**Reply by taismassaro** [reply]
It's actually quite confusing because I seem to be getting an error that was supposedly fixed in 2020 (!) according to #11892 🤔

**Reply by taismassaro** [reply]
@shilman update: after reading [this comment](https://github.com/storybookjs/storybook/pull/12085/files#diff-d1ae31d46b72bc0f8e58c34eded5eeeea109fe6247de6ad9f41901b64e259078R135-R147) in the code I realised the problem was that, because I was running both storybooks at the same time with turborepo, the package one was initializing _after_ the composition one, which caused the check for `isPublic` to fail. when running them separately one after the other it worked, so thanks for well documented code! 

maybe this could be better documented in the public docs? happy to contribute if that's the case.

**Reply by shilman** [reply]
@taismassaro great job figuring it out! @jonniebigodes any thoughts on where that might fit in the documentation?

**Reply by ranisalt** [reply]
What now that it got removed in v8?

**Reply by shilman** [reply]
@ranisalt There's a file called `index.json` that gets automatically created since v7.

Also, FWIW, a v8 storybook can compose an older version of storybook. But an older version of storybook cannot compose a newer version.

**Reply by ranisalt** [reply]
Are these files supposed to be equivalent? They don't share the same structure. The `index.json` file does not contain `parameters`, as `stories.json` used to.

We don't use it for composing, we use it to detect what stories contain changed components in a pull request in order to run snapshots.

**Comment by Timbo925** [comment]
I had also had problems with running 2 static storybook pages in the same docker container using nginx to server both. I ended up copying the stories.json from sub-book as a static file in main-book.

````
main-book (the storybook served on example.com)
sub-book (the storybook served on exmaple.com/sub-book referenced in main-book)
`````

#### Build Steps

1. Add `features: {buildStoriesJson: true},` to your `main.js` file in both storybooks.
2. In sub-book project run `build-storybook && cp .storybook/metadata.json storybook-static/metadata.json` to build the storybook-static folder. metadata.json is an empty {} json file. (without is was fetching a html page for some reason).
3. in main-book add the following to `main.js`. Then run `npm run build-storybook`
````js
staticDirs: [
    { from: '../../sub-book/storybook-static/stories.json', to: '/main-book/stories.json' },
],
refs: {
 platform: {
    title: 'sub-book',
    url: '/sub-book',
  },
}
````
4. Dockerfile for nginx container copies over both static storybooks. main-book is exposed on default path `/`, sub-book on `/sub-book`.
````Dockerfile
COPY ./main-book/storybook-static /usr/share/nginx/html/main-book
COPY ./sub-book/storybook-static /usr/share/nginx/html/sub-book
````

Seems abit like a roundabout way but by copying the `story.json` file over to the main-book and then building the static page, it seems like all the references will be correct without sub-book needing to be deployed. Then you only need to expose the static storybooks on the corresponding locations and storybook is happy.

Hopefully this is helpful for someone.

**Reply by mokone91** [reply]
im also facing with metadata.json fetching problem, there is no such file in buld result, but storybook requesting it for some reason.

**Reply by AntonyFagundez** [reply]
I'm having the same issue, to fixed in my case is make a dummy `metadata.json`. Because the problem is finding the file `metadata.json`,
metadata.json
```json
{}
```

```Dockerfile
COPY ./workaround-composition/metadata.json ./metadata.json
```

Dependecies used: 
```json
{
    "@storybook/addon-a11y": "^6.5.13",
    "@storybook/addon-docs": "^6.5.13",
    "@storybook/addon-essentials": "^6.5.13",
    "@storybook/addon-jest": "^6.5.13",
    "@storybook/addon-links": "^6.5.13",
    "@storybook/addons": "^6.5.13",
    "@storybook/react": "^6.5.13",
    "@storybook/react-docgen-typescript-plugin": "^1.0.1",
    "@storybook/builder-webpack5": "^6.5.13",
    "@storybook/manager-webpack5": "^6.5.13",
    "webpack": "^5.75.0",
    "@storybook/theming": "^6.5.13",
    "storybook-addon-outline": "^1.4.2",
    "storybook-addon-react-docgen": "^1.2.42",
    "eslint-plugin-storybook": "^0.6.7",
    "@storybook/addon-measure": "^6.5.13",
}
```

**Comment by MrBra1nwash** [comment]
Hey folks. Is there any way to make `stories.json` to be created without a build of the storybook?
I am using storybook `8.6.7` with a composition:
```
    refs: {
      first: {
        title: 'First',
        url: 'http://localhost:6010/',
      },
      second: {
        title: 'Second',
        url: 'http://localhost:6011/',
      },
    },
```

And when I run this composed storybook `locally` (without building) it works well and all stories are presented on the screen. But the console is full of errors like:
```
GET http://localhost:6010/stories.json 404 (Not Found)
GET http://localhost:6010/metadata.json 404 (Not Found)
GET http://localhost:6011/stories.json 404 (Not Found)
```

Also `http://localhost:6010/stories.json` returns 404 by direct access too. I think these files are not generated at all without the build. Do you have any plans to resolve it?

**Reply by dablorhbryan-chwy** [reply]
For anyone who also had this question, I'm guessing its now called `index.json`.

**Reply by shilman** [reply, marked as answer]
It's now called `index.json` and you can generate it by calling `storybook build` or `storybook index` (much faster because it only generates the index file and doesn't build the rest of the Storybook)

Edit: If you don't need the file on disk it's also served by `storybook dev`

**Reply by levrik** [reply]
Why is this file actually not automatically served by `storybook dev`? Do I really have to build it manually? Seems weird and unintuitive for local development.

**Reply by shilman** [reply]
@levrik It's also automatically served by `storybook dev`. Updated.

**Reply by ekeijl** [reply]
> It's now called `index.json` and you can generate it by calling `storybook build` or `storybook index` (much faster because it only generates the index file and doesn't build the rest of the Storybook)
> 
> Edit: If you don't need the file on disk it's also served by `storybook dev`

@shilman I'm confused why I'm seeing 404 errors for `stories.json` and `metadata.json`. By looking at the origin of the request, I can see it's coming from [these lines of code](https://github.com/storybookjs/storybook/blob/a0f880055c321926907e2fbcefa7e4ab7de939af/code/core/src/manager-api/modules/refs.ts#L236-L254).

From this code it seems `stories.json` is not optional?

I'm using `storybook build` to build the individual Storybooks. The build output directory only contains `index.json` and the network request for this file is successful, but the other two files are missing.

Currently using Storybook v9.0.15

**Reply by shilman** [reply]
Only `index.json` is produced in 9.x. `stories.json` was produced in older Storybooks. `metadata.json` was never produced.

**Reply by ekeijl** [reply]
> Only `index.json` is produced in 9.x. `stories.json` was produced in older Storybooks. `metadata.json` was never produced.

That is indeed what I'm seeing, but Storybook is still requesting `stories.json` and `metadata.json` as well, which is causing 404 errors. 

I'm trying to resolve this issue, but I can't tell whether this is a bug in Storybook or if I'm doing something wrong.

**Reply by antoinebarbier022** [reply]
Hello I have the same issue than @MrBra1nwash with "storybook": "10.0.8"

**Reply by shilman** [reply]
Storybook's composition feature first tries to pull `stories.json` and then tries to pull `index.json`. The order should probably be switched. The 404 should is an implementation detail and should not show up to the user unless the view the browser devtools.

EDIT: It does them in parallel.
